### PR TITLE
add abaplint.json configuration

### PR DESCRIPTION
**Pull request by larshp**
see https://github.com/SAP/code-pal-for-abap/issues/187

and https://github.com/abaplint/abaplint/blob/master/docs/ci/README.md for how to run automatically

image

### CODE REVIEW

**Comment by lucasborin** [comment]
I'll install the Github App in our repository.

File: abaplint.json [hunk 1]
```diff
@@ -0,0 +1,330 @@
...
+      "files": "/src/**/*.*"
+    },
+    {
+      "url": "https://github.com/abapGit/abapGit",
+      "folder": "/deps2",
+      "files": "/src/**/*.*"
+    }
+  ],
+  "syntax": {
+    "version": "v754",
```
[the remark this answers is not in the document]
Comment (larshp) [reply abaplint.json]: ahh, yea, updated in https://github.com/SAP/code-pal-for-abap/pull/188/commits/0d572c86e3e6635fee11c8c2aedb31b07676fe1e

**Comment by lucasborin** [comment]
@estevao-schultz-neto-SAP, @eugen-guenther-sap : Should we disable most of the checks and keep only the syntax compatibility in the first version?

**Comment by larshp** [comment]
I have disabled all the rules that gave errors, ie. only finding with this configuration is the syntax error mentioned in https://github.com/SAP/code-pal-for-abap/issues/187

File: abaplint.json [hunk 2]
```diff
@@ -0,0 +1,330 @@
...
+    },
+    {
+      "url": "https://github.com/abapGit/abapGit",
+      "folder": "/deps2",
+      "files": "/src/**/*.*"
+    }
+  ],
+  "syntax": {
+    "version": "v740sp08",
+    "errorNamespace": "^(Z|Y|LCL_|TY_|LIF_)",
```
Comment (lucasborin) [inline abaplint.json:24]: The project is under customer namespace (Y*). Can you please remove the `Y` from the `errorNamespace`?

**Review by lucasborin — CHANGES_REQUESTED** [review]

File: abaplint.json (hunk 2 above)
Comment (larshp) [reply abaplint.json:24]: The pattern in errorNamespace is the objects that abaplint should report errors for

With this configuration, errors are issued if a YCL_ is not found, or a ZIF_ is not found, https://github.com/SAP/code-pal-for-abap/blob/8d6d5a8b80108ceee4b376a4b2e70078cdf673b5/src/foundation/y_code_pal_service.clas.abap#L85

**Review by lucasborin — APPROVED** [review]

**Comment by lucasborin** [comment]
Many thanks for your contribution @larshp :)

**Comment by larshp** [comment]
let me know if it doesnt work out, I can also open a PR with a github actions setup for it, but note it doesn't work for forked pull requests

**Pull request merged**